### ISSUE TITLE
find_dns.py: line 40: cannot concatenate 'str' and 'int' objects

### ISSUE THREAD

**Issue by BikeMaker** [codingo/Reconnoitre#22]
Running Reconnoitre with the following options in Kali Rolling yields a concatenation TypeError:
```
# ./reconnoitre.py -t 192.168.0.0/24 -o ../../../Findings/ --pingsweep --dns --services --snmp
  __
|"""\-=  RECONNOITRE
(____)      An OSCP scanner by @codingo_

[#] Performing ping sweep
[!] ../../../Findings didn't exist and has been created.
[+] Writing discovered targets to: ../../../Findings/targets.txt
[+] Performing ping sweep over 192.168.0.0/24

...snip...

      [=] Found DNS service running on: 192.168.0.14
Traceback (most recent call last):
  File "./reconnoitre.py", line 89, in <module>
    main()
  File "./reconnoitre.py", line 66, in main
    find_dns(arguments.target_hosts, arguments.output_directory, arguments.quiet)
  File "/root/git/Reconnoitre/reconnoitre/find_dns.py", line 40, in find_dns
    results += 1
TypeError: cannot concatenate 'str' and 'int' objects

Line 40 of find_dns.py is incrementing 'results' by 1:


            if ("53/tcp" in line) and ("open" in line):
                print("      [=] Found DNS service running on: %s" % (ip_addres$
                output_file.write("[*] Found DNS service running on: %s\n" % (i$
                output_file.write("   [>] %s\n" % (line))
                output_targets.write("%s" % (ip_address))
        >>>     results += 1     <<<

```

The problem here is that in find_dns.py, at line 10, you're first casting 'results' as 0, then later at line 29 you set it to the output from subprocess.check_host() which returns as encoded bytes by default, but most definitely non-int in pretty much any case but error, which you later try to increment at line 40 like an int, which of course TypeErrors and traceback is fired.   

I'd normally submit a pull for this but am mobile atm, so just wanted to submit the details so you can throw in a quick fix.  Clearly you're meaning for a different var to be used either in the check_output() return or in the increment by 1 line.  

To note, I quickly tested changing the incremented var to something else which fixed the traceback and resolved the issue fwiw in the field. 

Cheers, great work on this!  Appreciate you open-sourcing it.

**Comment by codingo** [comment]
Thank-you for this. I'm tied up in an engagement for the next couple of weeks so I may not get to this one straight away but I'll be sure to update this issue once everything is resolved!

**Comment by smileronin** [comment]
Also, How can we speed up the detail service scan, because Scan job is very slowly stage of detail scan process, The scan is not proggresing.  Maybe can do  bypass the dns check? For example, Maybe you can add the --ignore-dns-check parameter.

# python reconnoitre.py -t 192.168.183.16 -o /OSCP/LAB/ --services

|"""\-=  RECONNOITRE
(____)      An OSCP scanner by @codingo_

[+] Testing for required utilities on your system.
[#] Performing service scans
[*] Loaded single target: 192.168.183.16
[+] Creating directory structure for 192.168.183.16
   [>] Creating scans directory at: /OSCP/LAB/192.168.183.16/scans
   [>] Creating exploit directory at: /OSCP/LAB/192.168.183.16/exploit
   [>] Creating loot directory at: /OSCP/LAB/192.168.183.16/loot
   [>] Creating proof file at: /OSCP/LAB/192.168.183.16/proof.txt
[+] Starting quick nmap scan for 192.168.183.16
mass_dns: warning: Unable to determine any DNS servers. Reverse DNS is disabled. Try using --system-dns or specify valid servers with --dns-servers
[+] Writing findings for 192.168.183.16
   [>] Found FTP service on 192.168.183.16:21
   [>] Found HTTP service on 192.168.183.16:80
[*] TCP quick scans completed for 192.168.183.16
[+] Starting detailed TCP/UDP nmap scans for 192.168.183.16
mass_dns: warning: Unable to determine any DNS servers. Reverse DNS is disabled. Try using --system-dns or specify valid servers with --dns-servers
....

**Comment by codingo** [comment]
Tagging this as help wanted as this project is a bit lower on my list, if somebody wants to take a crack at this I would very happily merge it over!

**Comment by codingo** [comment]
Fixed in #43
